pkg/config/config.go

```diff
@@ -171,6 +175,10 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return fmt.Errorf("bad auth config in defaults section: user/password and PAT authentication are mutually exclusive")
 	}
 
+	if c.Defaults.AutoResolve == true && c.Defaults.AutoResolveState == "" {
+		return fmt.Errorf("bad config in defaults section: auto_resolve_state must be defined when auto_resolve is set to true")
```

Comment:
What if `AutoResolveState` is defined at a receiver level? You cannot make this check here, but in the receivers.

Reply:
Good point. I moved this check in the receivers section.